bundles/org.openhab.binding.meater/src/main/java/org/openhab/binding/meater/internal/api/MeaterRestAPI.java

```diff
@@ -0,0 +1,169 @@
+            request.content(new StringContentProvider(json), JSON_CONTENT_TYPE);
+
+            logger.debug("HTTP POST Request {}.", request.toString());
+
+            ContentResponse httpResponse = request.send();
+            if (httpResponse.getStatus() != HttpStatus.OK_200) {
+                throw new MeaterException("Failed to login " + httpResponse.getContentAsString());
+            }
+            // Fetch JWT
+            json = httpResponse.getContentAsString();
+            JsonObject jsonObject = JsonParser.parseString(json).getAsJsonObject();
+            JsonObject childObject = jsonObject.getAsJsonObject("data");
+            this.authToken = childObject.get("token").getAsString();
+            this.userId = childObject.get("userId").getAsString();
+        } catch (InterruptedException | TimeoutException | ExecutionException | JsonParseException e) {
+            Thread.currentThread().interrupt();
+            throw new MeaterException(e);
+        }
+    }
+
+    private String getFromApi(String uri) throws MeaterException {
+        try {
+            for (int i = 0; i < MAX_RETRIES; i++) {
+                try {
+                    Request request = httpClient.newRequest(API_ENDPOINT + uri).method(HttpMethod.GET);
+                    request.header(HttpHeader.AUTHORIZATION, "Bearer " + authToken);
+                    request.header(HttpHeader.ACCEPT, JSON_CONTENT_TYPE);
+                    request.header(HttpHeader.CONTENT_TYPE, JSON_CONTENT_TYPE);
+                    request.header(HttpHeader.ACCEPT_LANGUAGE, acceptLanguage);
+
+                    ContentResponse response = request.send();
+                    String content = response.getContentAsString();
+                    logger.trace("API response: {}", content);
+
+                    if (response.getStatus() != HttpStatus.OK_200) {
+                        logger.debug("getFromApi failed, HTTP status: {}", response.getStatus());
+                        login();
+                    } else {
+                        return content;
+                    }
+                } catch (TimeoutException e) {
+                    logger.debug("TimeoutException error in get: {}", e.getMessage());
+                }
+            }
+            throw new MeaterException("Failed to fetch from API!");
+        } catch (InterruptedException | MeaterException | ExecutionException e) {
+            Thread.currentThread().interrupt();
+            throw new MeaterException(e);
+        }
+    }
+
+    public <T> T getDevices(Class<T> dto) throws MeaterException, JsonSyntaxException {
+        String uri = DEVICES;
+        String json;
+
+        try {
+            json = getFromApi(uri);
+        } catch (MeaterException e) {
+            throw new MeaterException(e);
+        }
+        if (json.isEmpty()) {
+            throw new MeaterException("JSON from API is empty!");
+        } else {
+            return gson.fromJson(json, dto);
```

Comment:
Consider wrapping exception here so caller doesn't **have to** deal with `JsonSyntaxException`:
```suggestion
            try {
                return gson.fromJson(json, dto);
            } catch (JsonSyntaxException e) {
                throw new MeaterException("Error parsing JSON", e);
            }
```
(see comment for MeaterException additional constructor overload)

Reply:
Fixed!